src/sonic-bgpcfgd/bgpcfgd/managers_device_global.py

```diff
@@ -0,0 +1,106 @@
+        :param table: name of the table in the db
+        """
+        self.directory = common_objs['directory']
+        self.cfg_mgr = common_objs['cfg_mgr']
+        self.constants = common_objs['constants']
+        self.tsa_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.isolate.conf.j2")
+        self.tsb_template = common_objs['tf'].from_file("bgpd/tsa/bgpd.tsa.unisolate.conf.j2")
+        super(DeviceGlobalCfgMgr, self).__init__(
+            common_objs,
+            [],
+            db,
+            table,
+        )
+
+    def set_handler(self, key, data):
+        log_debug("DeviceGlobalCfgMgr:: set handler")
+        """ Handle device tsa_enabled state change """
+        if not data:
+            log_err("DeviceGlobalCfgMgr:: data is None")
+            return False
+
+        if "tsa_enabled" in data:
+            self.cfg_mgr.commit()
+            self.cfg_mgr.update()
+            self.isolate_unisolate_device(data["tsa_enabled"])
+            self.directory.put(self.db_name, self.table_name, "tsa_enabled", data["tsa_enabled"])
+            return True        
+        return False
+
+    def del_handler(self, key):
+        log_debug("DeviceGlobalCfgMgr:: del handler")
+        return True
+
+    def check_state_and_get_tsa_routemaps(self, cfg):
+        """ API to get TSA route-maps if device is isolated"""
+        cmd = ""
+        if self.directory.path_exist("CONFIG_DB", "BGP_DEVICE_GLOBAL", "tsa_enabled"):
+            tsa_status = self.directory.get_slot("CONFIG_DB", "BGP_DEVICE_GLOBAL")["tsa_enabled"]
+            if tsa_status == "true":
+                cmds = cfg.replace("#012", "\n").split("\n")
+                log_notice("DeviceGlobalCfgMgr:: Device is isolated. Applying TSA route-maps")
+                cmd = self.get_tsa_routemaps(cmds)            
+        return cmd
+
+    def isolate_unisolate_device(self, tsa_status):
+        """ API to get TSA/TSB route-maps and apply configuration"""
+        cmd = "\n"
+        if tsa_status == "true":
+            log_notice("DeviceGlobalCfgMgr:: Device isolated. Executing TSA")
+            cmd += self.get_tsa_routemaps(self.cfg_mgr.get_text())
+        else:
+            log_notice("DeviceGlobalCfgMgr:: Device un-isolated. Executing TSB")
+            cmd += self.get_tsb_routemaps(self.cfg_mgr.get_text())
+
+        self.cfg_mgr.push(cmd)
+        log_debug("DeviceGlobalCfgMgr::Done")
+
+    def get_tsa_routemaps(self, cmds):
+        if not cmds:
+            return ""
+
+        route_map_names = self.__extract_out_route_map_names(cmds)
+        return self.__generate_routemaps_from_template(route_map_names, self.tsa_template)
+
+    def get_tsb_routemaps(self, cmds):
+        if not cmds:
+            return ""
+
+        route_map_names = self.__extract_out_route_map_names(cmds)
+        return self.__generate_routemaps_from_template(route_map_names, self.tsb_template)
+
+    def __generate_routemaps_from_template(self, route_map_names, template):
```

Comment:
The function of this mgr is very similar with TSA and TSB under the sonic-buildimage/dockers/docker-fpm-frr/, should we deprecate them?

Reply:
Yes, TSA/B script under sonic-buildimage/dockers/docker-fpm-frr will no longer be invoked and can be deprecated after this change